### ISSUE TITLE
Support installation from local repositories using foundryup

### ISSUE THREAD

**Issue by dmfxyz** [gakonst/foundry#929]
### Component

Foundryup

### Describe the feature you would like

Foundryup provides easy and repeatable installations of nightly, branched/versioned, and forked Foundry repositories.  Currently, it is only possible to install `forge` and `cast` from remote repositories/tarballs.

For both Foundry contributors and users, it would be convenient to enable `foundryup` to install from local repositories. This would simplify the testing of local changes across multiple projects and generally align local development with the install method used for remote codebases and official releases. 

In short, I propose a simple capability for local installations to be added to Foundryup. If there is agreement, I can clean-up my personal implementation and open it in a PR for review and modification.

### Additional context

Discussed briefly in Telegram with @gakonst

**Comment by onbjerg** [comment]
Any reason `cargo install --path .` isn't sufficient?

**Comment by dmfxyz** [comment]
@onbjerg It definitely gets the job done, but I am biased towards consistency, especially given that it can be accomplished without too many lines added to foundryup. From the perspective of a newcomer to the foundry ecosystem (which I most certainly am), having everything within foundryup would be nice.

**Comment by gakonst** [comment]
One way to do this would be symlinking the binaries under target/release of the specified repo to your $PATH I think?

**Comment by dmfxyz** [comment]
> One way to do this would be symlinking the binaries under target/release of the specified repo to your $PATH I think?

Yes, this is the approach I'm trying now. Originally I had symlinked to a new $AUTHOR folder in $FOUNDRY_DIR then installed from there. It worked but it made the control flow in the remote repo block over complicated. I think this can be done in a very short `if` block that's executed if a `--local-repo` value is passed. Will put up a draft PR tonight PT.

**Comment by dmfxyz** [comment]
@gakonst Low priority, but whenever you have a chance for feedback there's a PR open with the symlink implementation #949

**Comment by dmfxyz** [comment]
Feature merged.
